app/src/adapters/db/clients/postgres_client.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import os
+import urllib.parse
+from typing import Any
+
+import psycopg2
+import sqlalchemy
+import sqlalchemy.pool as pool
+
+from src.adapters.db.client import DBClient
+from src.adapters.db.config import DBConfig, get_db_config
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresDBClient(DBClient):
+    """
+    An implementation of a DBClient for connecting to a Postgres DB
+    as configured by parameters passed in from the db_config
+    """
+
+    def __init__(self, db_config: DBConfig | None = None) -> None:
```

Comment:
this makes me think that `DBConfig` is specific to `PostgresDBClient` so maybe we should rename it to `PostgresDBConfig`? but happy to defer that convo for now